aws/aws-toolkit-vscode
# Clean-up of Credentials Status Bar Item code and functionality

## Description

Changes related to the StatusBar Item that relates to the toolkit's active credentials.

Behavior changes:
- instead of hiding the item when there are no credentials in use, the text is updated to reflect this state
  ![image](https://user-images.githubusercontent.com/39839589/71370095-a1831800-2561-11ea-82fe-1227eaaba08e.png)
- item now has a tooltip
- item can now be clicked on to change the active credentials

Code changes:
- removed unnecessary interface+class wrapping the status bar item because it is never accessed after initialization
- StatusBar item text is now populated based on the credentials changed event data


## Motivation and Context

In preparation for some new credentials handling code, some of the existing credentials related code flows are being tidied up in separate, isolated changes.

## Testing

* Change credentials, see status bar text updates
* see that status bar shows expected text when there are no active credentials
* hover over status bar, verify that tooltip shows as expected
* clicking on status bar item prompts user to select credentials

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
